lib/ansible/modules/network/infinity/infinity.py

```diff
@@ -0,0 +1,650 @@
+# !/usr/bin/python
+# meiliu@fusionlayer.com
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+
+has_url_parse = True
+try:
+    # python3
+    import urllib.request as urllib_request
+    import urllib.parse as urllib_parse
+    from urllib.request import AbstractHTTPHandler
+    has_url_parse = True
+except ImportError as e:
+    # python2
+    has_url_parse = False
+    # print(" using python2 librarary")
+    import urllib2 as urllib2_request
+    import urllib
+    import base64
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+
+
+DOCUMENTATION = """
+module: infinity
+short_description: manage Infinity IPAM using  Web API
+description:
+  - Manage Infinity IPAM using Web API
+version_added: "1.0"
+author:
+  - "Meirong Liu"
+options:
+  server:
+    description:
+      - Infinity IP
+    required: True
+  username:
+    description:
+      -  username to access Infinity
+      - The user must have Rest API privileges
+    required: True
+  password:
+    description:
+      - Infinity password
+    required: True
+  action:
+    description:
+      - Action to perform
+    required: True
+    choices:  [ 'reserve_next_available_ip', 'release_ip' , 'delete_network',
+    'add_network' ,  'reserve_network', 'release_network', 'get_network_id'],
+
+  network_id:
+    description:
+      - Network ID
+      - Must be indicated as a network_id
+    required: False
+    default: False
+  ip_address:
+    description:
+      - IP Address for reservation or for release
+    required: False
+    default: False
+  network_address:
+    description:
+      - nework address  CIDR format (e.g 192.168.310.0)
+    required: False
+    default: False
+ network_size:
+    description:
+      - nework bitmask (e.g. 255.255.255.220)or CIDR format (e.g  /26)
+    required: False
+    default: False
+ network_name:
+    description:
+      - nework name
+    required: False
+    default: False
+ network_location:
+    description:
+        the parent network id for a given network
+    required: False
+    default: -1
+ network_type:
+    description:
+      - network type defined by     Infinity
+    required: False
+    "choices": ['lan', 'shared_lan', 'supernet']
+    default: "lan"
+ network_family:
+    description:
+      - network family defined by  Infinity, e.g. IPv4, IPv6 , Dual stack
+    required: False
+    "choices": ['4', '6', 'dual']
+    default: "4"
+
+"""
+RETURN = """
```

Comment:
Have a look at the `RETURN` block in another module along side the documentation at http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#return-block